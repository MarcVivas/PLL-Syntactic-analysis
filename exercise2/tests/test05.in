// Test logical expressions with ? and :

// Test simple expression with ?
1 + (1 > 0 ? 2 : 3);
// expected output: 3

// Test expression with nested ?
1 + (1 > 0 ? (2 > 1 ? 3 : 4) : 5);
// expected output: 4

// Test simple expression with :
1 - (1 < 0 ? 2 : 3);
// expected output: -2

// Test expression with nested :
654564 + (1 > 0 ? 2 : (2 < 1 ? 3 : 4));
// expected output: 654566

// Test expression with both ? and :
8 * (1 > 0 ? (2 >= 2 ? 3 : 4) : (2 <= 2 ? 5 : 6));
// expected output: 24

// Test expression with nested ? and :
4 + (1-7 > 0 ? (2-4 > 1 ? (3 == 3 ? 4-3 : 5+4) : 6*3) : (2+44 != 1 ? 7*1 : 8*8));
// expected output: 11

// Expressions using variables
a = 423*4/2; b = 423; c = 122/2; d = ((43+2) - (44 + 3))*2;
// Values: a = 846; b = 423; c = 61; d = -4
r = a > b ? c : d > c ? d : b < c ? b + d : a + b + c + d;
r;
// Expected output: 61
