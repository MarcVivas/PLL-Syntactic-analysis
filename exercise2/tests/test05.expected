3
4
-2
654566
24
11
61
a = 846
b = 423
c = 61
d = -4
r = 61
